core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/RowDenormalizerConfig.java

```diff
@@ -0,0 +1,161 @@
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/**
+ * Config for RowDenormalizer Aggregator Plugin
+ */
+public class RowDenormalizerConfig extends AggregatorConfig {
+
+  public static final String CHECK_ALIAS = "NO_ALIAS";
+  @Description("Key Field which will be used to de-normalize the raw data.")
+  private final String keyField;
+  @Description("Name of the field in input record that contains output fields to be included in denormalized output.")
+  private final String fieldName;
+  @Description("Name of the field in input record that contains values for output fields to be included " +
+    "in denormalized output.")
+  private final String fieldValue;
+  @Description("List of the output fields with its alias to be included in denormalized ouptut.")
+  private final String outputFields;
+  @Name("schema")
+  @Nullable
+  private final String schema;
+
+  @VisibleForTesting
+  RowDenormalizerConfig(String keyField, String fieldName, String fieldValue, String outputFields, String schema) {
+    this.keyField = keyField;
+    this.fieldName = fieldName;
+    this.fieldValue = fieldValue;
+    this.outputFields = outputFields;
+    this.schema = schema;
+  }
+
+  /**
+   * Returns name of the keyfield given as input by user.
+   *
+   * @return key field
+   */
+  String getKeyField() {
+    return keyField;
+  }
+
+  /**
+   * Returns name of the input field 'FieldName' given as input by user
+   *
+   * @return field name
+   */
+  String getFieldName() {
+    return fieldName;
+  }
+
+  /**
+   * Returns name of the input field 'FieldValue' given as input by user
+   *
+   * @return field value
+   */
+  String getFieldValue() {
+    return fieldValue;
+  }
+
+  /**
+   * Fetches the output fields or output field alias if provided, to build the output schema.
+   *
+   * @return List of output fields
+   */
+  List<String> getOutputFields() {
+    List<String> fields = new ArrayList<String>();
+    for (String field : Splitter.on(',').trimResults().split(outputFields)) {
+      String[] value = field.split(":");
+      if (value.length == 1) {
+        fields.add(value[0]);
+      } else {
+        fields.add(value[1]);
+      }
+    }
+    return fields;
+  }
+
+  /**
+   * Returns the output schema
+   *
+   * @return schema
+   */
+  String getSchema() {
+    return schema;
+  }
+
+  /**
+   * Creates a map for output field and its alias, if present.
+   *
+   * @return Map of output fields and its alias
+   */
+  Map<String, String> getOutputFieldMappings() {
+    Map<String, String> outputFieldMappings = new HashMap<String, String>();
+    for (String field : Splitter.on(',').trimResults().split(outputFields)) {
+      String[] value = field.split(":");
+      if (value.length == 1) {
```

Comment:
This is good idea. But, currently we are accepting alias for the output fields from user itself as per the requirement.
Let me know if this needs to be changed, accordingly i will make the changes in WIKI as well as code and documentation.